src/clj/puppetlabs/services/jruby_pool_manager/jruby_core.clj

```diff
@@ -229,11 +229,7 @@
   "Flush all the current JRubyInstances so that the pool can be shutdown
   without any instances being active."
   [pool-context]
-  (let [on-complete (promise)]
-    (log/debug "Beginning flush of JRuby pools for shutdown")
-    (jruby-agents/send-flush-pool-for-shutdown! pool-context on-complete)
-    @on-complete
-    (log/debug "Finished flush of JRuby pools for shutdown")))
+  (jruby-agents/flush-pool-for-shutdown! pool-context))
```

Comment:
Yeah, I agree with that (mentioned in a comment on the `drain-and-refill-pool!` function) - good catch.